bundles/org.openhab.binding.ipcamera/README.md

```diff
@@ -0,0 +1,1312 @@
+If you prefer to use your own firewall instead, you can also choose to make the `IP_WHITELIST` equal "DISABLE" (the default since the feature also needs a valid SERVER_PORT set) to turn this feature off and then all internal IP's will have access.
+
+There are multiple ways to get a moving picture, to use them just enter the url into any browser using ``http://192.168.xxx.xxx:SERVER_PORT/name.format`` replacing the name.format with one of the options that are listed below:
+
++ **ipcamera.m3u8** HLS (Http Live Streaming) which uses h264 compression. 
+This can be used to cast to Chromecast devices, or can display video in many browsers (some browsers require a plugin to be installed). Please understand that this format due to the way it works will give you lag behind realtime, more on this below.
++ **ipcamera.mjpeg** whilst needing more bandwidth, it is far more compatible for displaying in a wider range of UI's and browsers. It is normally 1 second or less behind real-time.
+Ffmpeg can be used to create this stream if your camera does not create one for you, but this uses more CPU. 
+A lot of cameras limit the resolution in this format, so consider using HLS, autofps.mjpeg, or snapshots.mjpeg instead which will be in a higher resolution.
++ **snapshots.mjpeg** is a special mjpeg stream created from the cameras snapshots that are taken at the polling rate.
++ **autofps.mjpeg** This requires a camera that has a motion alarm to be turned on or it will only send a picture every 8 seconds. You can also use the ``externalMotion`` channel to change the framerate.
+This feature is designed to keep data traffic to your mobile devices as low as possible by automatically sending 1fps when motion is occurring, but only 1 picture every 8 seconds when the picture has no motion.
+The idea is to not send lots of pictures if the picture has not changed as doing so only eats up your data plan.
++ **ipcamera.gif** This is small in size and very compatible and handy to use in push notifications, Pushover, Telegram, or emails. You can cast it which can be handy to show a moving picture that keeps repeating on a Google/Nest home hub or your wall mounted tablet. 
++ MP4 recordings can be created by the binding and FFmpeg, more on this below.
+
+This forum thread has examples of how to use snapshots and streams in a sitemap.
+<https://community.openhab.org/t/ip-camera-how-to-clickable-thumbnail-overview-in-sitemaps-that-opens-up-to-a-larger-view/77990>
+
+If you use Habpanel, then these widgets are worth checking out.
+<https://community.openhab.org/t/custom-widget-camera-clickable-thumbnails-that-open-a-stream/101275>
+
+
+
+## MP4 Recordings
+
+The binding can use FFmpeg to create a recording to a file.
+To do this:
+
++ Consider setting the String channel that is called `mp4Filename` to a date and time stamp in a format that you like, or leave the channel empty for the filename to default to `ipcamera.mp4`.
```

Comment:
Are these Channels missing above?

Reply:
mp4Filename is a channel and I use it daily. 
ipcamera.mp4 is not a channel, but is the default name it will give a recording if the above channel is an empty string.

Reply:
I have started a nicer list of all channels which is probably what you meant by this comment.